File: book/fridgets/java/fridgets/src/fridgets/Fridget.java
package fridgets;

import java.awt.Dimension;
import java.awt.event.KeyEvent;
import java.awt.event.MouseEvent;
import java.util.Optional;
import nz.sodium.*;

public abstract class Fridget {
    public static class Output {
        public Output(
                Cell<Drawable> drawable,
                Cell<Dimension> desiredSize,
                Stream<Long> sChangeFocus) {
            this.drawable = drawable;
            this.desiredSize = desiredSize;
            this.sChangeFocus = sChangeFocus;
        }
        public Cell<Drawable> drawable;
        public Cell<Dimension> desiredSize;
        public Stream<Long> sChangeFocus;
    }
    public Fridget(Lambda5<
            Cell<Optional<Dimension>>, Stream<MouseEvent>,
            Stream<KeyEvent>, Cell<Long>, Supply, Output> reify_) {
        this.reify_ = reify_;
    }
    private final Lambda5<
            Cell<Optional<Dimension>>, Stream<MouseEvent>,
            Stream<KeyEvent>, Cell<Long>, Supply, Output> reify_;
    public final Output reify(
            Cell<Optional<Dimension>> size,
            Stream<MouseEvent> sMouse, Stream<KeyEvent> sKey,
            Cell<Long> focus, Supply idSupply) {
        return reify_.apply(size, sMouse, sKey, focus, idSupply);
    }
}

File: book/fridgets/c#/Fridgets/Fridget.cs
using System;
using System.Windows;
using Sodium.Frp;
using Sodium.Functional;

namespace Fridgets
{
    public abstract class Fridget
    {
        public class Output
        {
            public Output(Cell<DrawableDelegate> drawable, Cell<Size> desiredSize, Stream<long> sChangeFocus)
            {
                this.Drawable = drawable;
                this.DesiredSize = desiredSize;
                this.SChangeFocus = sChangeFocus;
            }

            public Cell<DrawableDelegate> Drawable { get; }
            public Cell<Size> DesiredSize { get; }
            public Stream<long> SChangeFocus { get; }
        }

        private readonly Func<Cell<Maybe<Size>>, Stream<MouseEvent>, Stream<KeyEvent>, Cell<long>, Supply, Output> reify;

        protected Fridget(Func<Cell<Maybe<Size>>, Stream<MouseEvent>, Stream<KeyEvent>, Cell<long>, Supply, Output> reify)
        {
            this.reify = reify;
        }

        public Output Reify(Cell<Maybe<Size>> size, Stream<MouseEvent> sMouse, Stream<KeyEvent> sKey, Cell<long> focus, Supply idSupply)
        {
            return this.reify(size, sMouse, sKey, focus, idSupply);
        }
    }
}